packages/@react-stately/calendar/src/useCalendarState.ts

```diff
@@ -249,3 +285,11 @@ export function useCalendarState(props: CalendarStateOptions): CalendarState {
     }
   };
 }
+
+function unitDuration(duration: DateDuration) {
```

Comment:
i don't quite follow what this does, I see it's used in subtract and add above, but seems like it might have multiple keys in it, in which case, what happens? should it advance 1 week and 1 day if both those keys are in visibleDuration? or what happens if the value was like 3 for months: 3

Reply:
Yeah it's kinda like a unit vector, which is why it's named the way it is. Just converts all the units to 1. When advancing with page up/page down, we wanted to move by one "unit" at a time rather than by a whole page in case e.g. multiple months are displayed. However, we support other units besides months, e.g. weeks, or days (see useCalendar stories). So this accounts for that.

> should it advance 1 week and 1 day if both those keys are in visibleDuration?

yes, that's currently what would happen. it would be a pretty weird case if someone set that as their visibleDuration though.

> or what happens if the value was like 3 for months: 3

it would advance by 1 month.

Reply:
ahhh, ok, yeah, that would a pretty weird thing to display. thanks